### PR TITLE
feat: filter out MySQL federated components' emitted statements

### PR DESCRIPTION
**Pull request by chunshao90**
## Rationale
Some mysql syntax is not supported, such as `show variables`, and user errors will be returned.

## Detailed Changes
* Introduce `server/src/federated.rs` to filter some mysql syntax and return success. `server/src/federated.rs` inspired by  [greptime](https://github.com/GreptimeTeam/greptimedb/blob/702ea32538a99e2d163fb1fbd3e75b1ce4ec4232/src/servers/src/mysql/federated.rs).
Support following sql.
```
select @@version_comment
select version()
select 1
```
* Modify `scripts/license-header.txt`.

## Test Plan
Add some test and paas CI.

### CODE REVIEW

**Review by evenyag — DISMISSED** [review]
Rest LGTM

**Review by waynexia — DISMISSED** [review]
LGTM. 

BTW, this patch contains too many unrelated changes 😢

File: server/src/mysql/service.rs [hunk 1]
```diff
@@ -104,7 +104,7 @@ impl MysqlService {
 
                     let rt = runtimes.read_runtime.clone();
                     rt.spawn(AsyncMysqlIntermediary::run_on(
-                        MysqlWorker::new(proxy,  timeout),
+                        MysqlWorker::new(proxy,addr, timeout),
```
[the remark this answers is not in the document]
Comment (chunshao90) [reply server/src/mysql/service.rs]: > Why did the CI still pass?

In the `tokio::select!` macro, cargo fmt does not format the code...

File: server/src/session.rs [hunk 2]
```diff
@@ -0,0 +1,223 @@
...
+    pub fn new(client_addr: Option<SocketAddr>, channel: Channel) -> Self {
+        Self {
+            client_addr,
+            channel,
+        }
+    }
+}
+
+#[derive(Debug, PartialEq)]
+#[allow(dead_code)]
```
[the remark this answers is not in the document]
Comment (chunshao90) [reply server/src/session.rs:120]: > Do we have a plan to support different channels and dialects? If not, we could remove some unused codes.

Want to support `pgsql` dialect later.

**Review by evenyag — APPROVED** [review]

**Pull request merged**